src/mpyl/artifacts/build_artifacts.py

```diff
@@ -1,85 +1,62 @@
-
-            for artifact_path in os.listdir(self.cache_folder):
-                shutil.copytree(
-                    src=self.cache_folder / artifact_path,
-                    dst=artifact_path,
-                    dirs_exist_ok=True,
-                )
-        else:
-            self.logger.info(f"Branch '{branch}' has no remote to pull from")
-
-    def push(self, branch: str) -> None:
-        if self.artifact_repo.does_branch_exist(branch_name=branch):
-            if self.artifact_repo.get_branch != branch:
-                self.artifact_repo.checkout_branch(branch_name=branch)
-        else:
-            if self.artifact_repo.does_branch_exist(branch_name=branch, remote=True):
-                self.logger.info(f"Fetching branch '{branch}' from remote")
-                self.artifact_repo.fetch_branch(branch_name=branch)
-                self.artifact_repo.checkout_branch(branch_name=branch)
-            else:
-                self.logger.info(f"Creating new branch '{branch}'")
-                self.artifact_repo.create_branch(branch_name=branch)
-
-        shutil.rmtree(self.cache_folder, ignore_errors=True)
-
-        for artifact_path in self.get_build_artifacts_paths():
-            shutil.copytree(
-                src=artifact_path,
-                dst=self.cache_folder / artifact_path,
-                dirs_exist_ok=True,
-            )
-
-        if self.artifact_repo.has_changes():
-            self.logger.info("Committing and pushing all artifacts")
-            self.artifact_repo.stage(CACHE_FOLDER_NAME)
-            self.artifact_repo.commit(f"Add artifacts for {branch}")
-            self.artifact_repo.push()
+        with TemporaryDirectory() as tmp_repo_dir:
+            with Repository.from_clone(
+                config=self.artifact_repo_config, repo_path=Path(tmp_repo_dir)
+            ) as artifact_repo:
+                if artifact_repo.does_branch_exist(branch_name=branch):
+                    self.logger.info(f"Fetching branch '{branch}' from remote")
+                    artifact_repo.checkout_branch(branch_name=branch)
+                self.logger.info(f"Branch {branch} does not exist in remote")
+                artifact_repo.
+
+    def push(self, branch: str, file_paths: list[Path]) -> None:
+        with TemporaryDirectory() as tmp_repo_dir:
+            repo_path = Path(tmp_repo_dir)
+            with Repository.from_clone(
+                config=self.artifact_repo_config, repo_path=repo_path
+            ) as artifact_repo:
+                branch_exists = artifact_repo.does_branch_exist(branch_name=branch)
+                if branch_exists:
+                    self.logger.info(f"Fetching branch '{branch}' from remote")
+                    artifact_repo.checkout_branch(branch_name=branch)
+                else:
+                    artifact_repo.create_branch(branch_name=branch)
+
+                shutil.rmtree(repo_path, ignore_errors=True)
```

Comment:
Won't the pr now include the deletion of the readme and gitignore files that are in main? I guess it's not the biggest deal for our purposes, but that's what I tried to avoid with the extra 'cache' folder

Reply:
Yeah, good point. I think it's useful to have a target folder within the artifact repo for other reasons as well. In case of argocd you may have different pr/acc/prod folders in the same repo